aws-lc-rs/benches/kem_benchmark.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+use aws_lc_rs::{test, test_file};
+use criterion::{criterion_group, criterion_main, Criterion};
+
+#[allow(non_camel_case_types)]
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+enum Algorithm {
+    KYBER512_R3,
+}
+
+impl From<&str> for Algorithm {
+    fn from(value: &str) -> Self {
+        match value {
+            "KYBER512_R3" => Algorithm::KYBER512_R3,
+            _ => panic!("Unrecognized algorithm: '{value}'"),
+        }
+    }
+}
+
+pub struct KemConfig {
+    algorithm: Algorithm,
+    public_key: Box<[u8]>,
+    secret_key: Box<[u8]>,
+    ciphertext: Box<[u8]>,
+}
+
+impl KemConfig {
+    fn new(algorithm: &str, public_key: &[u8], secret_key: &[u8], ciphertext: &[u8]) -> Self {
+        KemConfig {
+            algorithm: Algorithm::from(algorithm),
+            public_key: Vec::from(public_key).into(),
+            secret_key: Vec::from(secret_key).into(),
+            ciphertext: Vec::from(ciphertext).into(),
+        }
+    }
+}
+
+macro_rules! benchmark_kem {
+    ( $pkg:ident ) => {
+        paste::item! {
+                mod [<$pkg _benchmarks>]  {
+
+                    use $pkg::kem;
+
+            use crate::{KemConfig, Algorithm};
+            use kem::{KemPrivateKey, KemPublicKey, KemAlgorithm, KYBER512_R3};
+
+            fn algorithm(config: &KemConfig) -> &'static KemAlgorithm {
+                match config.algorithm {
+                    Algorithm::KYBER512_R3 => &KYBER512_R3,
+                }
+            }
+
+            pub fn new_private_key(config: &KemConfig) -> KemPrivateKey {
+                KemPrivateKey::new(algorithm(config), &config.secret_key).unwrap()
+            }
+
+            pub fn new_public_key(config: &KemConfig) -> KemPublicKey {
+                KemPublicKey::new(algorithm(config), &config.public_key).unwrap()
+            }
+
+            pub fn keygen(config: &KemConfig) {
+                let private_key = KemPrivateKey::generate(algorithm(config)).unwrap();
+                let _public_key = private_key.compute_public_key().unwrap();
+            }
+
+            pub fn encapsulate(
+                public_key: &KemPublicKey,
+            ) {
+                public_key.encapsulate((), |ct, ss| {
+                    Ok((Vec::from(ct), Vec::from(ss)))
+                }).unwrap();
+            }
+
+            pub fn decapsulate(
+                config: &KemConfig,
+                secret_key: &KemPrivateKey,
+            ) {
+                let mut ciphertext = config.ciphertext.clone();
+                secret_key.decapsulate(&mut ciphertext, (), |ss| {
+                    Ok((Vec::from(ss)))
+                }).unwrap();
+            }
+        }
+                }
+    };
+}
+
+benchmark_kem!(aws_lc_rs);
+
+fn test_kem_keygen(c: &mut Criterion, config: &KemConfig) {
+    let bench_group_name = format!("KEM-keygen-{:?}", config.algorithm);
```

Comment:
Minor nit, same for the other ones below
```suggestion
    let bench_group_name = format!("KEM-{:?}-keygen", config.algorithm);
```